### PR TITLE
added watch statements to rebuild list Fixes #1107

### LINKED ISSUE #1107: Resizing of window creates bad layout in horizontal map widget
image

### PR DESCRIPTION
**Pull request by local-girl**
Added a watch to column width and column amount to trigger an list items reload. Looks a bit wonkey as it can only trigger after moving columns have loaded. Fixes #1107

### CODE REVIEW

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts [hunk 1]
```diff
@@ -324,6 +324,25 @@ export class MapListingController {
...
+        }, (newValue, oldValue) => {
+            if (newValue !== oldValue) {
+                // fixme: moving column load time hard coded
+                this.$timeout(() => {
+                    this.scrollToItem(this.$scope.selectedPath, true);
+                }, 550);
+            }
+        });
+        this.$scope.$watch(() => {
+            return this.map.getBounds()._southWest.lng;
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts]: Should be `getSouthWest()`. We should not use private properties.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts [hunk 2]
```diff
@@ -324,6 +324,25 @@ export class MapListingController {
...
+        }, (newValue, oldValue) => {
+            if (newValue !== oldValue) {
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts]: This is already done by `$watch`.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts [hunk 3]
```diff
@@ -324,6 +324,25 @@ export class MapListingController {
         this.$scope.$watch("items", () => {
             this.scrollToItem(this.$scope.selectedPath, false);
         });
+
+        var mcs = $element.parents(".moving-columns");
+        this.$scope.$watch(() => {
+            return mcs.find(".moving-column.is-show, .moving-column.is-collapse").length;
+        }, (newValue, oldValue) => {
+            if (newValue !== oldValue) {
+                // fixme: moving column load time hard coded
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts]: capital FIXME

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts [hunk 4]
```diff
@@ -324,6 +324,25 @@ export class MapListingController {
         this.$scope.$watch("items", () => {
             this.scrollToItem(this.$scope.selectedPath, false);
         });
+
+        var mcs = $element.parents(".moving-columns");
+        this.$scope.$watch(() => {
+            return mcs.find(".moving-column.is-show, .moving-column.is-collapse").length;
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts]: why not simply watch `$element.width()`?

Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Mapping/Mapping.ts]: another option would be to use `adhTopLevelState.on("movingColumns")`.

**Comment by xi** [comment]
I am not so sure about the watch expressions. Will have a closer look tomorrow.

**Comment by xi** [comment]
I changed the watch statement. @local-girl could you please review?

Some context on why I think this is better:

-   does not reply on columns. So it also works if we use this widget in another context, e.g. on its own.
-   more generic: Similar could be reused in other places.
-   simpler / less code.

**Comment by local-girl** [comment]
Yeah that's much better, merged!

**Pull request merged**